languages/dockerfile/tree-sitter/Parse_dockerfile_tree_sitter.ml

```diff
@@ -1032,9 +1032,23 @@ let source_file (env : env) (xs : CST.source_file) =
 (* Entry points *)
 (*****************************************************************************)
 
+(* tree-sitter-dockerfile parser requires a trailing newline.
+   Not sure if it's intentional but we add one to simplify user
+   experience.
+*)
+let ensure_trailing_newline str =
```

Comment:
Could you include a link to the upstream issue here for reference, especially if we think this is removable after a tree sitter version upgrade?

Reply:
Good idea. I added a comment with the relevant details.